### DISCUSSION TITLE
Combining LCOV Trace Files Information

### DISCUSSION THREAD

**Discussion by etaylork** [linux-test-project/lcov#266] [Q&A] [answered]
When combining two or more lcov trace files with the following command:

```
lcov -a lcov-unit.info -a lcov-e2e.info -o lcov.info
```

Will the output report take the aggregation of all the files to determine the line and functional coverage ? Asking because I have two different lcov trace files that are displaying different results where one file shows:
![image](https://github.com/linux-test-project/lcov/assets/22439603/9755b7d1-ae53-48dd-9aea-447e544dce2f)

And the other file shows:
![image](https://github.com/linux-test-project/lcov/assets/22439603/498f8374-874b-4cec-b88e-88f7c4f20645)

Once I combine the two files into one the data displays
79.4% for line coverage
69.2% for functional coverage

Can't find any exact documentation on what the processes of combining the coverage files looks like so I'm hoping to get some insight on here to better understand as my assumption was that if there is coverage for the same two files that when combining the output file would take the highest coverage of the two but that doesn't seem to be the case.

**Comment by henry2cox** [comment, marked as answer]
Yes.
Aggregation is essentially a set union operation:

- for coverpoints which exist in both input data sets, the 'hit; counts are  summed
- for coverpoints which exist in one or the other (but not both) - the (unique) point is added to the result.
This is legitimate - say, if your data sets reflect different compile options (e.g., `#ifdef`s or template specializations in your code).

Merging is obviously a problem/will cause misleading results if your input data does not correspond to the same source code version. 
lcov will check that for you if you like - see the `--version-script` and/or `--checksum` options.
genhtml will use the same `--version-script` callback to check that the displayed file matches the one used to extract data.

If the documentation seems unclear: please feel free to suggest an update/clarification.  That might help others as well.
